[ko] Update content/ko/docs/concepts/configuration/secret.md (#56766)
kubernetes/website: content/ko/docs/concepts/configuration/secret.md

Reviewer: ```suggestion
시크릿을 생성할 때, [Secret](/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/)
```

> When creating a Secret, you can specify its type using the `type` field of
the [Secret] (/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/)
resource, or certain equivalent `kubectl` command line flags (if available).

영어 원문에 맞춰 수정하였습니다.

| 빌트인 타입 | 사용처 |
|--------------|-------|
| `Opaque`     |  임의의 사용자 정의 데이터 |
| `kubernetes.io/service-account-token` | 서비스어카운트 토큰 |

> | `kubernetes.io/dockercfg` | 직렬화 된(serialized) `~/.dockercfg` 파일 |
> | `kubernetes.io/dockerconfigjson` | 직렬화 된 `~/.docker/config.json` 파일 |

Reviewer: ```suggestion
| `kubernetes.io/dockercfg` | 직렬화된(serialized) `~/.dockercfg` 파일 |
| `kubernetes.io/dockerconfigjson` | 직렬화된 `~/.docker/config.json` 파일 |
```

여기서`되다`는 피동의 뜻을 더하는 접미사이기 때문에 붙여쓰는 걸로 알고 있습니다.

[국립국어원 하다/되다 띄어쓰기](https://www.korean.go.kr/front/onlineQna/onlineQnaView.do?mn_id=216&qna_seq=326067&pageIndex=1&searchCondition=&searchKeyword=)

- `username`: 인증을 위한 사용자 이름
- `password`: 인증을 위한 암호나 토큰

위 두 키의 값은 모두 base64로 인코딩된 문자열이다. 대신
시크릿 매니페스트의 `stringData` 필드를 사용하여 평문 텍스트 콘텐츠(clear text content)를
제공할 수도 있다.

> 다음의 메니페스트는 기본 인증 시크릿의 예시이다.

Reviewer 2: '매니페스트'와 '메니페스트'가 혼용되어 있는데, '매니페스트'를 더 많이 사용하는 표현인 것 같아요 :)

Author: 매니페스트로 통일했습니다! 리뷰 감사합니다 :)

### TLS 시크릿

`kubernetes.io/tls` 시크릿 타입은 일반적으로 TLS를 위해 사용되는 인증서 및 관련된 키를 저장하기 위한 것이다.

TLS 시크릿의 일반적인 용도 중 하나는 [인그레스](/docs/concepts/services-networking/ingress/)에 대한 
전송 암호화(encryption in transit)를 구성하는 것이지만, 
다른 리소스와 함께 사용하거나 워크로드에서 직접 사용할 수도 있다. 
이 타입의 시크릿을 사용할 때는 `tls.key` 와 `tls.crt` 키가 
시크릿 구성의 `data` (또는 `stringData`) 필드에서 제공되어야 한다. 
그러나, API 서버가 각 키에 대한 값이 유효한지 실제로 검증하지는 않는다.

`stringData`를 사용하는 대신 `data` 필드를 사용하여
base64로 인코딩된 인증서와 개인 키를 제공할 수도 있다. 자세한 내용은
[시크릿 이름 및 데이터에 대한 제약 사항](#restriction-names-data)을 참고한다.

다음 YAML은 TLS 시크릿을 위한 구성 예시를 포함한다.

{{% code_sample language="yaml" file="secret/tls-auth-secret.yaml" %}}

TLS 시크릿 타입은 오직 편의만을 위해서 제공된다.
사용자는 TLS 인증에서 사용할 자격 증명을 위해 `Opaque` 타입을 생성할 수도 있다. 
그러나, 미리 정의되어 공개된 시크릿 타입(`kubernetes.io/tls`)을 사용하면 
프로젝트에서 시크릿 형식의 일관성을 유지하는 데 도움이 된다. API 서버는
이 타입의 시크릿에 필요한 키가 설정되어 있는지 검증한다.

`kubectl`을 사용하여 TLS 시크릿을 생성하려면 `tls` 하위 커맨드를 사용한다.

```shell
kubectl create secret tls my-tls-secret \
  --cert=path/to/cert/file \
  --key=path/to/key/file
```

공개/개인 키 쌍은 사전에 준비되어야 한다. `--cert`에 지정하는 공개 키 인증서는 .PEM 인코딩되어 있어야 하며
`--key`에 지정하는 개인 키와 일치해야 한다.

### 부트스트랩 토큰 시크릿

> `bootstrap.kubernetes.io/token` 시크릿 타입은 노드 부트스트랩 과정 에서 사용되는

Reviewer: ```suggestion
`bootstrap.kubernetes.io/token` 시크릿 타입은 노드 부트스트랩 과정에서 사용되는
```

> 쿠버네티스 매니페스트로서, 부트스트렙 토큰 시크릿은 다음과 유사할

Reviewer: ```suggestion
쿠버네티스 매니페스트로서, 부트스트랩 토큰 시크릿은 다음과 유사할
```

명시할 수 있다. 해당 `data` 와 `stringData` 필드는 선택적으로 명시할 수 있다.
`data` 필드의 모든 키(key)에 해당하는 값(value)은 base64로 인코딩된 문자열이어야 한다.
만약 사용자에게 base64로의 문자열 변환이 적합하지 않다면,
임의의 문자열을 값으로 받는 `stringData` 필드를 대신 사용할 수 있다.

`data` 및 `stringData`의 키는 영숫자 문자,
`-`, `_`, 또는 `.` 으로 구성되어야 한다. `stringData` 필드의 모든 키-값 쌍은 의도적으로
`data` 필드로 합쳐진다. 만약 키가 `data` 와 `stringData` 필드 모두에 정의되어
있으면, `stringData` 필드에 지정된 값이
우선적으로 사용된다.

#### 크기 제한 {#restriction-data-size}

개별 시크릿의 크기는 1 MiB로 제한된다. 
이는 API 서버 및 kubelet 메모리를 고갈시킬 수 있는 매우 큰 시크릿의 생성을 방지하기 위함이다. 
그러나, 작은 크기의 시크릿을 많이 만드는 것도 메모리를 고갈시킬 수 있다. 
[리소스 쿼터](/docs/concepts/policy/resource-quotas/)를 사용하여 
한 네임스페이스의 시크릿 (또는 다른 리소스) 수를 제한할 수 있다.

### 시크릿 수정하기

만들어진 시크릿은 [불변(immutable)](#secret-immutable)만 아니라면 수정될 수 있다.
시크릿 수정 방식은 다음과 같다.

* [`kubectl` 사용하기](/docs/tasks/configmap-secret/managing-secret-using-kubectl/#edit-secret)
* [설정 파일 사용하기](/docs/tasks/configmap-secret/managing-secret-using-config-file/#edit-secret)

[Kustomize 도구](/docs/tasks/configmap-secret/managing-secret-using-kustomize/#edit-secret)로
시크릿 내부의 데이터를 수정하는 것도 가능하지만, 이 경우 수정된 데이터를 지닌 새로운 `Secret` 오브젝트가 생성된다.

시크릿을 생성한 방법이나 파드에서 시크릿이 어떻게 사용되는지에 따라,
존재하는 `Secret` 오브젝트에 대한 수정은 해당 데이터를 사용하는 파드들에 자동으로 전파된다.

> 자세한 정보는 [마운트된 시크릿의 자동 업데이트](#마운트된-시크릿의-자동-업데이트)를 참고하라.

Reviewer: ```suggestion
시크릿을 생성한 방법이나 파드에서 시크릿이 어떻게 사용되는지에 따라,
존재하는 `Secret` 오브젝트에 대한 수정은 해당 데이터를 사용하는 파드들에 자동으로 전파된다.
자세한 정보는 [파드에서 시크릿을 파일처럼 사용하기](#using-secrets-as-files-from-a-pod)를 참고하라.
```

>Depending on how you created the Secret, as well as how the Secret is used in
your Pods, updates to existing `Secret` objects are propagated automatically to
Pods that use the data. For more information, refer to [Using Secrets as files from a Pod](#using-secrets-as-files-from-a-pod) section.

참조 헤더가 변경된 것 같습니다. 반영 부탁드립니다.

### 시크릿 사용하기

시크릿은 데이터 볼륨으로 마운트되거나 파드의 컨테이너에서 사용할
{{< glossary_tooltip text="환경 변수" term_id="container-env-variables" >}}로
노출될 수 있다. 또한, 시크릿은 파드에 직접 노출되지 않고,
시스템의 다른 부분에서도 사용할 수 있다. 예를 들어, 시크릿은
시스템의 다른 부분이 사용자를 대신해서 외부 시스템과 상호 작용하는 데 사용해야 하는
자격 증명을 보유할 수 있다.

특정된 오브젝트 참조(reference)가 실제로 시크릿 유형의 오브젝트를 가리키는지 확인하기 위해, 
시크릿 볼륨 소스의 유효성이 검사된다. 
따라서, 시크릿은 자신에 의존하는 파드보다 먼저 생성되어야 한다.

시크릿을 가져올 수 없는 경우 
(아마도 시크릿이 존재하지 않거나, 또는 API 서버에 대한 일시적인 연결 불가로 인해) 
kubelet은 해당 파드 실행을 주기적으로 재시도한다. 
kubelet은 또한 시크릿을 가져올 수 없는 문제에 대한 세부 정보를 포함하여 해당 파드에 대한 이벤트를 보고한다.

#### 선택적 시크릿 {#restriction-secret-must-exist}

파드에서 시크릿을 참조할 때는 다음 예시와 같이 해당 시크릿을
_선택 사항_ 으로 표시할 수 있다. 선택 사항 시크릿이
존재하지 않는 경우 쿠버네티스는 이를 무시한다.

{{% code_sample language="yaml" file="secret/optional-secret.yaml" %}}

기본적으로 시크릿은 필수이다. 모든 필수 시크릿이 사용 가능해지기 전에는
파드의 어떤 컨테이너도 시작되지 않는다.

> 파드가 시크릿의 특정 필수 키를 참조하고 해당 시크릿이 존재하지만 지정한 키가 존재하지 않는 경우
> 파드는 시작 과정에서 실패한다.

Reviewer: ```suggestion
파드가 선택 사항이 아닌 시크릿의 특정 키를 참조하고, 해당 시크릿은 존재하지만 지정한 키가 없는 경우 
파드는 시작 과정에서 실패한다.
```

> If a Pod references a specific key in a non-optional Secret and that Secret
does exist, but is missing the named key, the Pod fails during startup.

`non-optional`은 `key`가 아니라 `Secret`을 수식하는 걸로 이해하고 있습니다.
이에 맞춰 수정하였습니다.

시크릿의 업데이트는 API 감시 메커니즘(기본값), 유효 기간(time-to-live)이 정의된 캐시, 또는
각 kubelet 동기화 루프마다 클러스터 API 서버를 폴링하는 방식을 통해
전파될 수 있다.

따라서 시크릿이 업데이트되는 시점부터 새로운 키가 파드에 반영되는
시점까지의 전체 지연 시간은 kubelet 동기화 주기 + 캐시 전파 지연 시간만큼
길어질 수 있으며 여기서 캐시 전파 지연 시간은 선택한 캐시 유형에 따라
달라진다(앞 문단에서 나열한 순서대로 감시 전파 지연 시간,
구성된 캐시 TTL 또는 직접 폴링의 경우 0이다).

### 시크릿을 환경 변수 형태로 사용하기

파드에서 {{< glossary_tooltip text="환경 변수" term_id="container-env-variables" >}} 형태로
시크릿을 사용하려면 다음과 같이 한다.

1. 파드 명세의 각 컨테이너에 대해 사용하려는 각
  시크릿 키마다 `env[].valueFrom.secretKeyRef` 필드를 사용하는
  환경 변수를 추가한다.
1. 프로그램이 지정된 환경 변수에서 값을 찾도록 
   이미지 및/또는 커맨드 라인을 수정한다.

자세한 지시 사항은 
[시크릿 데이터를 사용하여 컨테이너 환경 변수 정의](/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data)를 참고한다.

파드에서 환경 변수 이름으로 사용할 수 있는 문자에는
[제약 사항](/docs/tasks/inject-data-application/define-environment-variable-container/#using-environment-variables-inside-of-your-config)이
있다는 점에 유의해야 한다. 일부 키가 규칙을 충족하지 않으면
해당 키는 컨테이너에서 사용할 수 없지만 파드는 정상적으로 시작된다.

### 컨테이너 이미지 풀 시크릿 {#using-imagepullsecrets}

비공개 저장소에서 컨테이너 이미지를 가져오고 싶다면, 
각 노드의 kubelet이 해당 저장소에 인증을 수행하는 방법을 마련해야 한다. 
이를 위해 _이미지 풀 시크릿_ 을 구성할 수 있다. 
이러한 시크릿은 파드 수준에 설정된다.

> #### imagePullSecrets 사용하기
> 
> `imagePullSecrets` 필드는 동일한 네임스페이스의 시크릿에 대한 참조 목록이다.
> `imagePullSecrets` 를 사용하여 도커(또는 다른 컨테이너) 이미지 레지스트리 비밀번호가 포함된 시크릿을 kubelet에 전달할 수 있다. 
> kubelet은 이 정보를 사용해서 파드를 대신하여 프라이빗 이미지를 가져온다.
> `imagePullSecrets` 필드에 대한 자세한 정보는 [PodSpec API](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core)를 참고한다.
> 
> ##### imagePullSecret 수동으로 지정하기
> 
> [컨테이너 이미지](/docs/concepts/containers/images/#파드에-imagepullsecrets-명시) 문서에서 
> `imagePullSecrets`를 지정하는 방법을 배울 수 있다.
> 
> ##### imagePullSecrets가 자동으로 연결되도록 준비하기
> 
> 수동으로 `imagePullSecrets` 를 생성하고, 
> 서비스어카운트에서 이들을 참조할 수 있다. 
> 해당 서비스어카운트로 생성되거나 기본적인 서비스어카운트로 생성된 모든 파드는 
> 파드의 `imagePullSecrets` 필드를 가져오고 서비스어카운트의 필드로 설정한다. 
> 해당 프로세스에 대한 자세한 설명은 
> [서비스어카운트에 ImagePullSecrets 추가하기](/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account)를 참고한다.
> 
> ### 스태틱 파드에서의 시크릿 사용 {#restriction-static-pod}
> 
> {{< glossary_tooltip text="스태틱(static) 파드" term_id="static-pod" >}}에서는 
> 컨피그맵이나 시크릿을 사용할 수 없다.
> 

Reviewer: ```suggestion
#### imagePullSecrets 사용하기

`imagePullSecrets` 필드는 동일한 네임스페이스의 시크릿에 대한 참조 목록이다.
`imagePullSecrets` 를 사용하여 도커(또는 다른 컨테이너) 이미지 레지스트리 비밀번호가 포함된 시크릿을 kubelet에 전달할 수 있다. 
kubelet은 이 정보를 사용해서 파드를 대신하여 프라이빗 이미지를 가져온다.
`imagePullSecrets` 필드에 대한 자세한 정보는 [PodSpec API](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core)를
참고한다.

##### imagePullSecret 수동으로 지정하기

[컨테이너 이미지](/docs/concepts/containers/images/#파드에-imagepullsecrets-명시) 문서에서 
`imagePullSecrets`를 지정하는 방법을
배울 수 있다.

##### imagePullSecrets가 자동으로 연결되도록 준비하기

수동으로 `imagePullSecrets` 를 생성하고, 서비스어카운트에서 이들을 참조할 수 있다. 
해당 서비스어카운트를 사용하여 생성되거나 해당 서비스어카운트가 기본적으로 적용되어 생성된 모든 파드는
파드의 `imagePullSecrets` 필드를 가져오고 서비스어카운트의 필드로 설정한다. 
해당 프로세스에 대한 자세한 설명은 
[서비스어카운트에 ImagePullSecrets 추가하기](/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account)를 참고한다.

### 스태틱 파드에서의 시크릿 사용 {#restriction-static-pod}

{{< glossary_tooltip text="스태틱(static) 파드" term_id="static-pod" >}}에서는 컨피그맵이나 시크릿을 사용할 수 없다.

```

>Any Pods created with that ServiceAccount or created with that ServiceAccount by default, will get their imagePullSecrets field set to that of the service account.

`598Line`에서 `created with that ServiceAccount by default`는 `기본적인 서비스어카운트로 생성된 모든 파드`가 아니라 `해당 ServiceAccount가 파드에 기본적으로 적용되는 경우`를 말하는 것으로 이해하고 있습니다. 이에 맞춰 수정하였습니다.

또한, 중간 중간에 개행 구조가 원문과 달라 맞춰 수정하였습니다.

쿠버네티스는 시크릿 오브젝트에 대해 약간의 추가적인 보호 조치를 적용한다.

시크릿은 종종 다양한 중요도에 걸친 값을 보유하며, 이 중 많은 부분이

> 쿠버네티스(예: 서비스어카운트 토큰)와 외부 시스템으로 단계적으로
> 확대될 수 있다. 개별 앱이 상호 작용할 것으로 예상되는 시크릿의 힘에 대해 추론할 수 있더라도

Reviewer: ```suggestion
쿠버네티스(예: 서비스어카운트 토큰)와 외부 시스템에서
권한 상승으로 이어질 수 있다. 개별 앱이 상호 작용할 것으로 예상되는 시크릿의 힘에 대해 추론할 수 있더라도
```

> many of which can cause escalations within Kubernetes (e.g. service account tokens) and to external systems.

여기서 `escalations`는 시스템으로부터 부여받은 권한보다 더 높은 권한을 획득할 수 있다는 권한 상승의 뜻으로 보입니다.
